server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -655,9 +660,13 @@ export class AgenticChatController implements ChatHandlers {
         triggerContext: TriggerContext,
         additionalContext: AdditionalContentEntryAddition[],
         chatResultStream: AgenticChatResultStream
-    ): Promise<GenerateAssistantResponseCommandInput> {
+    ): Promise<ChatCommandInput> {
         this.#debug('Preparing request input')
-        const profileArn = AmazonQTokenServiceManager.getInstance().getActiveProfileArn()
+        // Get profileArn from the service manager if available
+        let profileArn = undefined
+        if (this.#serviceManager instanceof AmazonQTokenServiceManager) {
```

Comment:
please implement the getActiveProfileArn in the serviceManager. In the AmazonQBaseServiceManager and AmazonQIAMServiceManager, do nothing 

Avoid having this `if` everywhere.